src/segment_evaluation/mod.rs

```diff
@@ -32,6 +32,7 @@ pub(crate) struct SegmentRules {
     kind: ValueKind,
 }
 
+// TODO: We should rename this to TargetingRules
```

Comment:
After this PR, we probably need to spend some time getting our naming straight here. This cost me quite some time today to understand the confusion again.